core/src/main/java/com/linecorp/armeria/client/AbstractEventLoopState.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client;
+
+import java.util.List;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import io.netty.channel.EventLoop;
+
+abstract class AbstractEventLoopState {
+
+    static AbstractEventLoopState of(List<EventLoop> eventLoops, int maxNumEventLoops,
```

Comment:
nit: How about making `interface EventLoopState` and extract common method for this, because a factory method which returns `abstract class` is a little strange for me 🙃

Reply:
https://github.com/line/armeria/pull/1886/commits/ae50467882dd6d568b3a4ece242eedde2fdb952a#diff-603fc6081c9252d8fcfee38b3fae0719R30
That was the implementation that but as @trustin said, this is package-private and it will make it complicated by introducing another interface, so just removed it. 😄

Reply:
Sorry, I should've caught review histories. 😅